### PR TITLE
Update osgi annotations in datastorage

### PR DESCRIPTION
**Pull request by PiteroS8**
## Description
I updated osgi annotations in datastorage

## Motivation and Context
Before, annotations was marked as deprecated

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the [code style](https://github.com/Cognifide/aet/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

File: core/datastorage/src/main/java/com/cognifide/aet/vs/mongodb/configuration/MongoDBClientConf.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  String ALLOW_AUTO_CREATE = "AllowAutoCreate";
+
+  boolean DEFAULT_AUTOCREATE_VALUE = false;
+
+  @AttributeDefinition(name = MONGO_URI, description =
+      "mongodb://[username:password@]host1[:port1][,host2[:port2],"
+          + "...[,hostN[:portN]]][/[database][?options]]", type = AttributeType.STRING)
+  String MongoURI() default DEFAULT_MONGODB_URI;
+
+  @AttributeDefinition(name = ALLOW_AUTO_CREATE, description = "Allows automatic creation of DB if set to true", type = AttributeType.BOOLEAN)
```
Comment (tkaik) [inline core/datastorage/src/main/java/com/cognifide/aet/vs/mongodb/configuration/MongoDBClientConf.java]: `name` should be set to the `label` value of old annotation

File: core/datastorage/src/main/java/com/cognifide/aet/vs/mongodb/configuration/MongoDBClientConf.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  String DEFAULT_MONGODB_URI = "mongodb://localhost";
+
+  String ALLOW_AUTO_CREATE_PROPERTY_NAME = "Allow automatic creation of DB";
+
+  String ALLOW_AUTO_CREATE_PROPERTY_DESCRIPTION = "Allows automatic creation of DB if set to true";
+
+  boolean DEFAULT_AUTOCREATE_VALUE = false;
+
+  @AttributeDefinition(name = MONGO_URI_PROPERTY_NAME, description = MONGO_URI_PROPERTY_DESCRIPTION, type = AttributeType.STRING)
+  String MongoURI() default DEFAULT_MONGODB_URI;
```
Comment (Skejven) [inline core/datastorage/src/main/java/com/cognifide/aet/vs/mongodb/configuration/MongoDBClientConf.java]: In Java usually methods names start with lowercase. Please adjust methods names in this interface.

Comment (PiteroS8) [reply core/datastorage/src/main/java/com/cognifide/aet/vs/mongodb/configuration/MongoDBClientConf.java]: There's a problem with that adjustment. `@AttributeDefinition` is case sensitive to keys from `*.cfg`
Example: `MongoURI=mongodb://localhost`

Comment (PiteroS8) [reply core/datastorage/src/main/java/com/cognifide/aet/vs/mongodb/configuration/MongoDBClientConf.java]: https://github.com/Cognifide/aet/issues/320

**Pull request merged**